gce_workload_cert_refresh/main.go

```diff
@@ -72,8 +72,12 @@ func getMetadata(key string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	defer res.Body.Close()
 
+	if res.StatusCode == 404 {
+		return nil, fmt.Errorf("HTTP 404")
+	}
+
+	defer res.Body.Close()
```

Comment:
you might be leaking this Body, move the defer closer to where it got allocated, inside the loop right after you test Do()'s returned error.

Reply:
good catch - i will update this in future PR. it is low risk right now as this program does not stay resident, but exits on completion.